server/repository/mongo/service_shifts.py

```diff
@@ -1,59 +1,78 @@
+        Adds new service shifts to the database.
+        
+        Args:
+            shift_data (list): A list of service shift dictionaries.
+        
+        Returns:
+            list: A list of inserted shift IDs.
         """
         for shift in shift_data:
-            shift.pop('_id', None)
-        self.db.service_shifts.insert_many(shift_data)
-
+            shift.pop("_id", None)  # Remove _id to let MongoDB generate it
+        result = self.collection.insert_many(shift_data)
+        return [str(shift_id) for shift_id in result.inserted_ids]
 
     def check_shift_overlap(self, shelter_id, shift_start, shift_end):
         """
-        checks if new shift overlaps with existing shifts for shelter
-        true if there is overlap and false otherwise
+        Checks if a new shift overlaps with an existing shift.
+        
+        Args:
+            shelter_id (int): The shelter ID.
+            shift_start (int): The start time of the new shift.
+            shift_end (int): The end time of the new shift.
+        
+        Returns:
+            bool: True if there is an overlap, False otherwise.
         """
-        overlapping = self.collection.find_one({
-            'shelter_id': shelter_id,
-            'shift_start': {'$lt': shift_end},
-            'shift_end': {'$gt': shift_start}
-        })
+        overlapping = self.collection.find_one(
+            {
+                "shelter_id": shelter_id,
+                "shift_start": {"$lt": shift_end},
+                "shift_end": {"$gt": shift_start},
+            }
+        )
         return overlapping is not None
 
-    def list(self, shelter=None):
+    def list(self, shelter_id=None, filter_start_after=None):
         """
-        Gets all service shifts in the database
-        returns list of all shift documents
+        Retrieves service shifts, optionally filtered by shelter_id and time.
+        
+        Args:
+            shelter_id (int, optional): The ID of the shelter.
+            filter_start_after (int, optional): The minimum start time.
+        
+        Returns:
+            list: A list of service shift objects.
         """
         db_filter = {}
-        if shelter:
-            db_filter['shelter_id'] = shelter
+        if shelter_id:
+            db_filter["shelter_id"] = shelter_id
+        if filter_start_after:
+            db_filter["shift_start"] = {"$gt": filter_start_after}
+
         service_shifts = [
-            ServiceShift.from_dict(i) for i in self.collection.find \
-            (filter=db_filter)
+            ServiceShift.from_dict(shift)
+            for shift in self.collection.find(db_filter)
         ]
         return service_shifts
-
-    def get_shifts(self, shift_ids):
-        """
-        gets multiple shifts by a list of ids
-        """
-        shifts = self.collection.find({'_id': {'$in': shift_ids}})
-        return [ServiceShift.from_dict(shift) for shift in shifts]
```

Comment:
Why did you delete this? I think we will want this, when retrieving information about multiple shifts

Reply:
@Taktar could you add that deleted code back in?